gapic/ads-templates/%namespace/%name/%version/%sub/services/%service/client.py.j2

```diff
@@ -175,24 +181,42 @@ class {{ service.client_name }}(metaclass={{ service.client_name }}Meta):
             client_options = ClientOptions.from_dict(client_options)
         if client_options is None:
             client_options = ClientOptions.ClientOptions()
+        
+        # Create SSL credentials for mutual TLS if needed.
+        use_client_cert = os.getenv("GOOGLE_API_USE_CLIENT_CERTIFICATE", "false")
+        if not use_client_cert in ["true", "false"]:
+            raise MutualTLSChannelError(
+                "Unsupported GOOGLE_API_USE_CLIENT_CERTIFICATE value. Accepted values: true, false"
+            )
+
+        ssl_credentials = None
+        is_mtls = False
+        if use_client_cert == "true":
+            if client_options.client_cert_source:
+                cert, key = client_options.client_cert_source()
+                ssl_credentials = grpc.ssl_channel_credentials(
+                    certificate_chain=cert, private_key=key
+                )
+                is_mtls = True
+            else:
+                creds = SslCredentials()
+                is_mtls = creds.is_mtls
+                ssl_credentials = creds.ssl_credentials if is_mtls else None
 
-        if transport is None and client_options.api_endpoint is None:
-            use_mtls_env = os.getenv("GOOGLE_API_USE_MTLS", "never")
+        # Figure out which api endpoint to use.
+        if client_options.api_endpoint is not None:
+            api_endpoint = client_options.api_endpoint
+        else:
+            use_mtls_env = os.getenv("GOOGLE_API_USE_MTLS_ENDPOINT", "auto")
```

Comment:
Similar to my above comment: I wish I'd pushed back on this earlier.
What we really want for `GOOGLE_API_USE_MTLS_ENDPOINT` is an Enum and the ability to convert from strings easily.
The _best_ thing to do would be to have some common utility classes in api-core with something like the following:
```python
import enum

class StringableEnum(enum.EnumMeta):
    def from_str(cls, s, default=None):
        return next((value for name, value in cls.__members__.items() if name == s), default)

class MTLSEndpointOption(enum.IntEnum, metaclass=StringableEnum):
    AUTO = 0
    NEVER = 1
    ALWAYS = 2
````
and then use it like
```python
use_mtls_env = MTLSEndpointOption.get(os.getenv("GOOGLE_API_USE_MTLS_ENDPOINT", "auto").upper())
if use_mtls_env == MTLSEndpointOption.NEVER:
...
```
This is way, way outside the scope for this review. I'll look into adding that support to api core or somewhere, and then file an issue to use that functionality here as part of a cleanup effort.

Reply:
sure. I will clean it up once the util function is added to api core.